client/src/pages/rollup/Show.js

```diff
@@ -82,6 +117,170 @@ const GQL_EMAIL_ROLLUP = gql`
+    display: "inline-block"
+  }
+  return (
+    <div className="scrollable-y">
+      <ContainerDimensions>
+        {({ width }) => (
+          <>
+            <ButtonToggleGroup value={orgType} onChange={setOrgType}>
+              <Button
+                value={Organization.TYPE.ADVISOR_ORG}
+                variant="outline-secondary"
+              >
+                {pluralize(Settings.fields.advisor.org.name)}
+              </Button>
+              <Button
+                value={Organization.TYPE.PRINCIPAL_ORG}
+                variant="outline-secondary"
+              >
+                {pluralize(Settings.fields.principal.org.name)}
+              </Button>
+            </ButtonToggleGroup>
+            <DailyRollupChart
+              width={width}
+              chartId={CHART_ID}
+              data={displayedGraphData}
+              onBarClick={setOrg}
+              tooltip={d => `
+              <h4>${d.org.shortName}</h4>
+              <p>Published: ${d.published}</p>
+              <p>Cancelled: ${d.cancelled}</p>
+              <p>Click to view details</p>
+            `}
+              barColors={barColors}
+            />
+          </>
+        )}
+      </ContainerDimensions>
+
+      <div className="graph-legend">
+        <div style={{ ...legendCss, background: barColors.published }} />{" "}
+        Published reports:&nbsp;
+        <strong>
+          {displayedGraphData.reduce((acc, org) => acc + org.published, 0)}
+        </strong>
+      </div>
+      <div className="graph-legend">
+        <div style={{ ...legendCss, background: barColors.cancelled }} />{" "}
+        Cancelled engagements:&nbsp;
+        <strong>
+          {displayedGraphData.reduce((acc, org) => acc + org.cancelled, 0)}
+        </strong>
+      </div>
+    </div>
+  )
+}
+
+Chart.propTypes = {
+  queryParams: PropTypes.object,
+  pageDispatchers: PageDispatchersPropType,
+  setOrg: PropTypes.func
+}
+
+const generateChartDataFromAllReports = (allReports, orgFilterUuid) => {
+  return allReports.reduce(
+    (acc, r) => {
+      if (r.advisorOrg) {
+        const topLevelAdvisorOrg = r.advisorOrg.ascendantOrgs[0]
+        // If reports are not filtered for organization show reports under the top level organization
+        const displayedAdvisorOrg = orgFilterUuid
+          ? r.advisorOrg
+          : topLevelAdvisorOrg
+        // Initialize the organization object if it is the first report belongs to the organization
+        if (!acc.advisorOrgReports[displayedAdvisorOrg.uuid]) {
+          acc.advisorOrgReports[displayedAdvisorOrg.uuid] = {
+            org: displayedAdvisorOrg,
+            published: 0,
+            cancelled: 0
+          }
+        }
+        if (r.state === Report.STATE.PUBLISHED) {
+          acc.advisorOrgReports[displayedAdvisorOrg.uuid].published++
+        }
+        if (r.state === Report.STATE.CANCELLED) {
+          acc.advisorOrgReports[displayedAdvisorOrg.uuid].cancelled++
+        }
```

Comment:
Code here and below does the same thing; extract this into a method and call that from both places.
Proposed change:
```
diff --git a/client/src/pages/rollup/Show.js b/client/src/pages/rollup/Show.js
index 375e0989e..39009655f 100644
--- a/client/src/pages/rollup/Show.js
+++ b/client/src/pages/rollup/Show.js
@@ -228,6 +228,21 @@ Chart.propTypes = {
   setOrg: PropTypes.func
 }
 
+const updateOrgReports = (orgReports, displayedOrg, reportState) => {
+  // Initialize the organization object if it is the first report belongs to the organization
+  const elem = (orgReports[displayedOrg.uuid] ??= {
+    org: displayedOrg,
+    published: 0,
+    cancelled: 0
+  })
+  if (reportState === Report.STATE.PUBLISHED) {
+    elem.published++
+  } else if (reportState === Report.STATE.CANCELLED) {
+    elem.cancelled++
+  }
+  return elem
+}
+
 const generateChartDataFromAllReports = (allReports, orgFilterUuid) => {
   return allReports.reduce(
     (acc, r) => {
@@ -237,42 +252,14 @@ const generateChartDataFromAllReports = (allReports, orgFilterUuid) => {
         const displayedAdvisorOrg = orgFilterUuid
           ? r.advisorOrg
           : topLevelAdvisorOrg
-        // Initialize the organization object if it is the first report belongs to the organization
-        if (!acc.advisorOrgReports[displayedAdvisorOrg.uuid]) {
-          acc.advisorOrgReports[displayedAdvisorOrg.uuid] = {
-            org: displayedAdvisorOrg,
-            published: 0,
-            cancelled: 0
-          }
-        }
-        if (r.state === Report.STATE.PUBLISHED) {
-          acc.advisorOrgReports[displayedAdvisorOrg.uuid].published++
-        }
-        if (r.state === Report.STATE.CANCELLED) {
-          acc.advisorOrgReports[displayedAdvisorOrg.uuid].cancelled++
-        }
+        updateOrgReports(acc.advisorOrgReports, displayedAdvisorOrg, r.state)
       }
       if (r.principalOrg) {
         const topLevelPrincipalOrg = r.principalOrg.ascendantOrgs[0]
-
         const displayedPrincipalOrg = orgFilterUuid
           ? r.principalOrg
           : topLevelPrincipalOrg
-
-        // Initialize the organization object if it is the first report belongs to the organization
-        if (!acc.principalOrgReports[displayedPrincipalOrg.uuid]) {
-          acc.principalOrgReports[displayedPrincipalOrg.uuid] = {
-            org: displayedPrincipalOrg,
-            published: 0,
-            cancelled: 0
-          }
-        }
-        if (r.state === Report.STATE.PUBLISHED) {
-          acc.principalOrgReports[displayedPrincipalOrg.uuid].published += 1
-        }
-        if (r.state === Report.STATE.CANCELLED) {
-          acc.principalOrgReports[displayedPrincipalOrg.uuid].cancelled += 1
-        }
+        updateOrgReports(acc.principalOrgReports, displayedPrincipalOrg, r.state)
       }
 
       return acc
```